es/sources_index_test.go

```diff
@@ -96,21 +98,98 @@ func (suite *SourcesIndexerSuite) TestSourcesIndex() {
 	})
 	suite.validateSourcesFullPath(indexNameEn, indexer, [][]string{[]string{source1UID, "t1", "t2"}, []string{source2UID, "t3", "t4"}})
 
-	//TBD add test for indexing with position (chapter)
+	//TBD add test for indexing with position (chapter)parentChapterPositionUID := suite.us(Source{Name: "test-name-3", ParentID: null.NewInt64(int64(consts.SRC_SHAMATI), true),
+
+	fmt.Printf("\n\n\nAdd source Shamati.\n\n")
+	parentChapterPosition := Source{Name: "Shamati"}
+
+	parentChapterPositionUID, parentChapterPositionID := suite.us(parentChapterPosition, consts.LANG_ENGLISH)
 
-	//TBD add test for indexing with description
+	consts.ES_SRC_PARENTS_FOR_CHAPTER_POSITION_INDEX[parentChapterPositionUID] = true
+	//r.Nil(indexer.SourceUpdate(parentChapterPositionUID))
+	//source2UID
+
+	fmt.Println("Validate adding source with file but without author - should not index.")
+	suite.usfc(parentChapterPositionUID, consts.LANG_ENGLISH)
+	//r.Nil(indexer.SourceUpdate(parentChapterPositionUID))
+
+	fmt.Println("Validate adding source with file and author and validate.")
+	suite.asa(Source{MDB_UID: parentChapterPositionUID}, consts.LANG_ENGLISH, mdbmodels.Author{Name: "Test Name 2", ID: 7, Code: "t5"}, true, true)
+	r.Nil(indexer.SourceUpdate(parentChapterPositionUID))
+	///
+	/// parentChapterPositionID = -1
+	/// First test for -1 position, is it doesn't indexed
+	chapterPositionUID, _ := suite.us(Source{Name: "test-name-3",
```

Comment:
you need to add the test that check it is actually not indexed (with validateFullNames)